code/parachain/frame/fnft/src/lib.rs

```diff
@@ -471,14 +473,15 @@ pub mod pallet {
 	}
 
 	impl<T: Config> FinancialNft<AccountIdOf<T>> for Pallet<T> {
-		/// TODO (vim): Assess the probability of collision in generating accounts
-		///   with collection id type u128 and instance id type u128 this definitely collides
-		///   because the seed is longer than the account ID
 		fn asset_account(
 			collection: &Self::CollectionId,
 			instance: &Self::ItemId,
 		) -> AccountIdOf<T> {
-			T::PalletId::get().into_sub_account_truncating((collection, instance))
+			// `into_sub_account_truncating()` gives us 20 bytes of space to create a seed.
+			// `blake2_128()` returns 16 bytes of data. Opperation will not create collisions.
+			T::PalletId::get().into_sub_account_truncating(sp_io::hashing::blake2_128(
+				&(collection, instance).encode(),
+			))
```

Comment:
The problem I ran into early with using the (u32, u64) was that our AccountId type in the test was u128 and not the standard AccountId32 we use in runtime. I'm changing this, so we don't have to call an expensive hash function here.

Reply:
9cfa1ab7effdedfe8c3a9da11155cf68c37779d2